typegate/tests/runtimes/wasmedge/wasmedge_sync_test.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+import { BasicAuth, tgDeploy, tgRemove } from "@typegraph/sdk/tg_deploy.js";
+import { gql, Meta } from "test-utils/mod.ts";
+import { testDir } from "test-utils/dir.ts";
+import { tg } from "./wasmedge.ts";
+import * as path from "std/path/mod.ts";
+import { connect } from "redis";
+import { S3Client } from "aws-sdk/client-s3";
+import { createBucket, listObjects, tryDeleteBucket } from "test-utils/s3.ts";
+import { assertEquals, assertExists } from "std/assert/mod.ts";
+
+const redisKey = "typegraph";
+const redisEventKey = "typegraph_event";
+
+async function cleanUp() {
+  using redis = await connect(syncConfig.redis);
+  await redis.del(redisKey);
+  await redis.del(redisEventKey);
+
+  const s3 = new S3Client(syncConfig.s3);
+  await tryDeleteBucket(s3, syncConfig.s3Bucket);
+  await createBucket(s3, syncConfig.s3Bucket);
+  s3.destroy();
+  await redis.quit();
+}
+
+const syncConfig = {
+  redis: {
+    hostname: "localhost",
+    port: 6379,
+    password: "password",
+    db: 1,
+  },
+  s3: {
+    endpoint: "http://localhost:9000",
+    region: "local",
+    credentials: {
+      accessKeyId: "minio",
+      secretAccessKey: "password",
+    },
+    forcePathStyle: true,
+  },
+  s3Bucket: "metatype-sync-test",
+};
+
+const cwd = path.join(testDir, "runtimes/wasmedge");
+const auth = new BasicAuth("admin", "password");
+
+Meta.test(
+  {
+    name: "WasmEdge Runtime typescript SDK: Sync Config",
```

Comment:
This seems very similar to the `typegraph_sync_test`, was a solution found to the leaks?

Reply:
Yeah, some of the things causing the leak were
- calling `engine.terminate()` would close the typegate without closing any redis connection, thus leaving unflushed resources.
- not closing all redis connection initialized during typegate init. So check if there are any unclosed resources.